core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -558,6 +566,11 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
       resourcesFileOpt, resourceProfileId)
   }
 
+  private def updateLogLevel(newLogLevel: String): Unit = {
+    if (newLogLevel != Utils.getLogLevel) {
+      Utils.setLogLevel(Level.toLevel(newLogLevel))
+    }
+  }
```

Comment:
rename to `setLogLevelIfNeeded` and move it to `Utils` you can then use it in `SparkContext` too